internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -423,6 +426,27 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		if rdy {
 			ready++
 		}
+
+		if err := r.composite.Render(ctx, cr, cd, comp.Spec.Resources[i]); err != nil {
+			log.Debug(errRenderCR, "error", err)
+			r.record.Event(cr, event.Warning(reasonCompose, err))
+			return reconcile.Result{RequeueAfter: shortWait}, nil
+		}
+	}
+
+	// Update composite spec and status with patches from composed resources
+	// DeepCopy composite to preserve rendered status
+	crSpecObject := cr.DeepCopyObject()
+	if err := r.client.Update(ctx, crSpecObject); err != nil {
+		log.Debug(errUpdate, "error", err)
+		r.record.Event(cr, event.Warning(reasonCompose, err))
+		return reconcile.Result{RequeueAfter: shortWait}, nil
+	}
+
+	if err := r.client.Status().Update(ctx, cr); err != nil {
```

Comment:
Have you confirmed this works? I would expect this update to always fail due to a stale `metadata.resourceVersion`. i.e.

1. `cr` is at `metadata.resourceVersion` `N`.
2. We make a deepcopy of `cr`.
3. We call `client.Update()`, passing in the deepcopy. This causes the API server to bump its expected resource version to `N+1`.
4. We call `client.Status().Update()`, passing in the original `cr`. Its `metadata.resourceVersion` is still `N`, so I would expect the API server to reject this update with an error like "the object has been modified; please apply your changes to the latest version and try again".

Reply:
If this spec update occurs, then later status updates will receive `resourceVersion` conflicts, but these occur when the spec has been updated, so this will lead to an another reconciliation of the resource and eventual consistency.

We discussed this some on slack - but because the composite needs both the spec and status updated here, the options are to update one sub-field and overwrite the resourceVersion for the other update, or handle a second reconcile. Choosing the path of a further reconcile leaves slightly clearer logic for the controller, so I've updated the PR and removed the explicit status update here as most* of the paths in the remaining reconcile logic leads to to updating the status regardless.